app/models/ZkKafka.scala

```diff
@@ -60,9 +60,10 @@ object ZkKafka {
         val path = makePath(applyBase(Seq(stormZkRoot, Some(root))) ++ Seq(Some(spout)) ++ Seq(Some(part)))
         // Use the first partition's data to build up info about the topology
         // Also, don't trust zk to have valid JSON, it may be null or something...
-        getZkData(path).map { zkData =>
-          val jsonState = new String(zkData)
-          val state = Json.parse(jsonState)
+        for {
```

Comment:
Comprehensions are a good fit for a lot of cases:
- easy to compose manyw map and flatMap operations together
- composing operations that require their bound variables to be in scope for the following operations (instead of nesting)
- sometimes the most compact and readable syntax for a particular series of operations (can look a lot like imperative style)

Map is usually more readable for single mapping operations, or very simple operations where you want to use placeholder syntax.

Largely it's a style choice guided by what results in the most clear and readable code, with compactness being a secondary concern, IMO.